src/Symfony/Component/Emoji/Emojis.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Emoji;
+
+use Symfony\Component\Emoji\Util\GzipStreamWrapper;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+final class Emojis
+{
+    /**
+     * Checks if an emoji exists.
+     */
+    public static function exists(string $emoji): bool
+    {
+        foreach (self::getEmojis() as $value) {
```

Comment:
this implementation is extemly subperformant: getEmoji turns the keys of an array into an iterator, then we scan the iterator. But checking an array by key is an O(1) operation while here we're using O(n)

Reply:
Yep sorry... i built the list in data/emojis.php seems better